x/aswap/msg.go

```diff
@@ -1,52 +1,29 @@
 package aswap
 
 import (
-	"github.com/iov-one/weave"
 	coin "github.com/iov-one/weave/coin"
 	"github.com/iov-one/weave/errors"
 	"github.com/iov-one/weave/migration"
 )
 
 func init() {
-	// Migration needs to be registered for every message introduced in the codec.
-	// This is the convention to message versioning.
 	migration.MustRegister(1, &CreateMsg{}, migration.NoModification)
 	migration.MustRegister(1, &ReleaseMsg{}, migration.NoModification)
 	migration.MustRegister(1, &ReturnSwapMsg{}, migration.NoModification)
 }
 
 const (
-	pathCreateSwap  = "aswap/create"
-	pathReleaseSwap = "aswap/release"
-	pathReturnSwap  = "aswap/return"
-
 	maxMemoSize int = 128
 	// preimage size in bytes
 	preimageSize int = 32
 	// preimageHash size in bytes
 	preimageHashSize int = 32
 )
 
-var _ weave.Msg = (*CreateMsg)(nil)
```

Comment:
I like these to help enforce the Msg interface here, rather than some random error when type casting in another file. Did you move these "assertions" somewhere else, or just delete them?

I feel when you make a struct with the sole purpose of fulfilling an interface, declaring that in a language the compiler understand is a "good thing"

Reply:
I was deleting all of them because I think there are enough uses of message and model interface to validate interface implementation. There is migration in the same file and handler and model registrations.
I used to put those everywhere before we had those checks.

Should they stay anyway?

Reply:
I quite like interface validation on top, it gives a nice indication of what needs to be implemented and show and error in the IDE if that's not the case. 

Very nice for newcomers too.